packages/grid/x-data-grid/src/hooks/features/export/useGridPrintExport.tsx

```diff
@@ -279,18 +321,28 @@ export const useGridPrintExport = (
         apiRef.current.forceUpdate();
       }
 
-      await updateGridColumnsForPrint(options?.fields, options?.allColumns);
+      await updateGridColumnsForPrint(
+        options?.fields,
+        options?.allColumns,
+        options?.includeCheckboxes,
+      );
+
+      const hasSelectedRows = apiRef.current.getSelectedRows().size > 0;
+      if (hasSelectedRows) {
```

Comment:
Changing the default behavior would be a breaking change.
Could you keep the default behavior and add the [`getRowsToExport` option](https://mui.com/x/react-data-grid/export/#exported-rows) to `printOptions`?
This way, we'll have the same API for CSV, Excel, and Print export 🙂

Reply:
I'm not sure what you mean, which default behavior is being changed? The `getRowsToExport` function is a method on the `GridPrintExportOptions` already, that's how I set it up initially in my first commit which is why it may not be obvious here, unless you mean something else. But [here's where that is being set](https://github.com/mui/mui-x/blob/0cd371544221552b23a04226f8efecdb5346803d/packages/grid/x-data-grid/src/models/gridExport.ts#L145) on the `printOptions`. I agree about trying to keep the API between all the export methods as similar as possible which is why I set it up this way.

Reply:
Gotcha, I understand now what you meant by default behavior. However, I don't agree that this should be the default behavior. I think that is unintuitive and does not maintain consistency with the other export options.

Currently, if I have rows selected and export via CSV only selected rows are exported. Same with Excel. Why would the user expectation for print be any different? As a long time user and instructor of MUI I can say that the typical user surely expects the print export to behave the same and I found it confusing when that was not the case. I'm not the only one as [another issue](https://github.com/mui/mui-x/issues/9824) was also raised regarding this.

If we were to leave the current behavior and rely solely on `getRowsToExport` for this functionality the solution to achieve the same effect of only exporting selected rows would look like this:
`slotProps={{ toolbar: { printOptions: { getRowsToExport: ({ apiRef }) => Array.from(apiRef.current.getSelectedRows().keys()) } }}}`

This is extremely unintuitive and requires far greater understanding of the internal workings of the component and of JavaScript itself (knowing that `.keys()` returns a map iterator and must be converted to an array using `Array.from()`). It's also extremely verbose for such a simple feature. I do not think this is an acceptable solution for the typical user.

I think the vast majority of users expect that if they have some rows selected then only those rows will be exported regardless of the export method. I think having different behavior between the different export methods only serves to confuse users and add more complexity to what should be quite a simple functionality.

I would like to hear others' opinions on this as well.

Reply:
> Gotcha, I understand now what you meant by default behavior. However, I don't agree that this should be the default behavior.

Regardless if we agree with you or not on that, we need to conform to [semver rules](https://semver.org/) to avoid breaking existing users' code (like any package on npm should do). So we're not going to include a change that breaks current code.

However, maybe there's an opportunity to refactor things a bit here, and allow passing `printOptions` from the root of the grid? This would make things more ergonomics for developpers.

```tsx
<DataGrid
  {...}
  printOptions={{ getRowsToExport: () => ... }}
/>
```

Reply:
> Regardless if we agree with you or not on that, we need to conform to [semver rules](https://semver.org/) to avoid breaking existing users' code (like any package on npm should do). So we're not going to include a change that breaks current code.

@romgrk - Fair enough. I understand that is a valid concern. I am earnestly curious about the original intention behind this behavior and the reasoning behind its discrepancy with the other export options if this is indeed an intended feature and not a bug. But oh well.

Accepting that, I would say that I think `printOptions` is fine where it is as a prop to the `GridToolbar` as that's where `csvOptions` and `excelOptions` live as well. Since they all concern export options and the export functionality lives within the `GridToolbar` I think that is all fine and doesn't necessitate moving up to a base prop of the `DataGrid`.

My issue remains more with how verbose the solution would be for developers to implement this functionality using solely `getRowsToExport` having to do something like: `getRowsToExport: ({ apiRef }) => Array.from(apiRef.current.getSelectedRows().keys())` if they wanted to only export selected rows via print.

Given that, I propose that I refactor my current solution to preserve the current functionality (not introduce the breaking change) and instead add another option to `printOptions` called something like `exportOnlySelected` that implements this same functionality. That would be a much simpler option for developers and require far less understanding of the internal workings of the DataGrid as opposed to the `getRowsToExport` solution above.

What do you think?

Reply:
@cherniavskii i agree with @zreecespieces, the print functionality now is not intuitive. The correct functionality would be like export csv or excel. When we select some rows at print export should be printed only selected, if nothing selected, print all. I personally need this fix asap or if you can tell me how to resolve it with a workaround it would be great.

Reply:
@cherniavskii no worries. I think that sounds great. I will update the PR in the morning 👍🏻

Reply:
PR has been updated based on this feedback. Current behavior is preserved, `getRowsToExport` and `includeCheckboxes` have been added to `printOptions` and documentation has been added for `printOptions` and using `getRowsToExport` to export only selected rows via print.

Let me know if you have any other feedback that needs to be addressed.